### PR TITLE
Fix fury bug

### PR DESCRIPTION
**Pull request by heliang666s**
## What is the purpose of the change

Fix fury bug 

https://github.com/apache/dubbo/issues/14536 

## Brief changelog

Fix FuryInputStream.java.By the way,add more tests.

## Verifying this change

XXXXX

Follow this checklist to help us incorporate your contribution quickly and easily:

- [x] Make sure there is a [GITHUB_issue](https://github.com/apache/dubbo/issues) field for the change (usually before
  you start working on it). Trivial changes like typos do not require a GITHUB issue. Your pull request should address
  just this issue, without pulling in other changes - one PR resolves one issue.
- [ ] Format the pull request title like `[Dubbo-XXX] Fix UnknownException when host config not exist #XXX`. Each commit
  in the pull request should have a meaningful subject line and body.
- [ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
- [ ] Write necessary unit-test to verify your logic correction, more mock a little better when cross module dependency
  exist. If the new feature or significant change is committed, please remember to add sample
  in [dubbo samples](https://github.com/apache/dubbo-samples) project.
- [ ] Run `mvn clean install -DskipTests=false` & `mvn clean test-compile failsafe:integration-test` to make sure
  unit-test and integration-test pass.
- [ ] If this contribution is large, please follow
  the [Software Donation Guide](https://github.com/apache/dubbo/wiki/Software-donation-guide).

### CODE REVIEW

**Comment by yunshen-Lee** [comment]
After testing, there are still issues with deserialization.
Caused by: java.lang.ArrayIndexOutOfBoundsException: 49
at org.apache.fury.resolver.MetaStringBytes.decode(MetaStringBytes.java:69) ~[fury-core-0.7.0.jar!/:0.7.0]
at org.apache.fury.resolver.ClassResolver.populateBytesToClassInfo(ClassResolver.java:1708) ~[fury-core-0.7.0.jar!/:0.7.0]
at org.apache.fury.resolver.ClassResolver.loadBytesToClassInfo(ClassResolver.java:1699) ~[fury-core-0.7.0.jar!/:0.7.0]
at org.apache.fury.resolver.ClassResolver.readClassInfoFromBytes(ClassResolver.java:1686) ~[fury-core-0.7.0.jar!/:0.7.0]
at org.apache.fury.resolver.ClassResolver.readClassInfo(ClassResolver.java:1608) ~[fury-core-0.7.0.jar!/:0.7.0]
at org.apache.fury.Fury.readRef(Fury.java:857) ~[fury-core-0.7.0.jar!/:0.7.0]
at org.apache.fury.Fury.deserializeJavaObjectAndClass(Fury.java:1225) ~[fury-core-0.7.0.jar!/:0.7.0]
at org.apache.fury.Fury.deserializeJavaObjectAndClass(Fury.java:1245) ~[fury-core-0.7.0.jar!/:0.7.0]

**Comment by songxiaosheng** [comment]
please fix conflict

**Comment by heliang666s** [comment]
> please fix conflict

have done

**Comment by AlbumenJ** [comment]
@chaokunyang PTAL

File: dubbo-serialization-extensions/dubbo-serialization-fury/src/main/java/org/apache/dubbo/common/serialize/fury/dubbo/FuryObjectInput.java [hunk 1]
```diff
@@ -40,7 +40,7 @@ public FuryObjectInput(Fury fury, MemoryBuffer buffer, InputStream input) {
 
   @Override
   public Object readObject() {
-    return fury.deserialize(new FuryInputStream(input, buffer.size()));
+    return fury.deserializeJavaObjectAndClass(new FuryInputStream(input, buffer.size()));
```
Comment (chaokunyang) [inline dubbo-serialization-extensions/dubbo-serialization-fury/src/main/java/org/apache/dubbo/common/serialize/fury/dubbo/FuryObjectInput.java:43]: Could we use `BlockedStreamUtils.deserialize(fury, output, obj);`  and `BlockedStreamUtils.serialize(fury, output, obj);` instead?  

`FuryInputStream` will buffer data internally for speed, which may read more data than needed

Comment (yunshen-Lee) [inline dubbo-serialization-extensions/dubbo-serialization-fury/src/main/java/org/apache/dubbo/common/serialize/fury/dubbo/FuryObjectInput.java:43]: We should not always create a new FuryInputStream here, but instead use the same one. We should create a FuryInputStream when constructing FuryObjectInput

Comment (chaokunyang) [reply dubbo-serialization-extensions/dubbo-serialization-fury/src/main/java/org/apache/dubbo/common/serialize/fury/dubbo/FuryObjectInput.java:43]: If the passed `InputStream` is used by dubbo after fury consumed more data, the usage in dubbo may fail. If we created `FuryInputStream`, we should use `FuryInputStream` in all places when using original `InputStream` just like we use `BufferedInputStream`

**Pull request merged**